app/models/admin.rb

```diff
@@ -0,0 +1,10 @@
+class Admin < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  has_many :given_kudos, class_name: 'Kudo', foreign_key: 'giver_id', dependent: :restrict_with_exception,
+         inverse_of: :admin
+  has_many :received_kudos, class_name: 'Kudo', foreign_key: 'receiver_id', dependent: :restrict_with_exception,
+            inverse_of: :admin
```

Comment:
Why do you create an association between `admin` and `kudo`? It is nod needed

also there are few mistakes, that you should learn from (but not correct as this part is not needed):
- `foreign_key` -> this is name of column where you store your admin_id in kudo, but you passed here columns that stores employee_id
-  `inverse_of` -> this should have name of another side of the same assosiation in Kudo model

Reply:
please check inverse_of in `employee` model